ethcore/src/tx_filter.rs

```diff
@@ -79,23 +79,15 @@ impl TransactionFilter {
 		match cache.entry((*parent_hash, sender)) {
 			Entry::Occupied(entry) => *entry.get() & tx_type != 0,
 			Entry::Vacant(entry) => {
-				let mut contract = self.contract.lock();
-				if contract.is_none() {
-					*contract = Some(Contract::new(self.contract_address));
-				}
-
-				let permissions = match &*contract {
-					&Some(ref contract) => {
-						contract.allowed_tx_types(
-							|addr, data| futures::done(client.call_contract(BlockId::Hash(*parent_hash), addr, data)),
-							sender,
-						).wait().unwrap_or_else(|e| {
-							debug!("Error callling tx permissions contract: {:?}", e);
-							tx_permissions::NONE
-						})
-					}
-					_ => tx_permissions::NONE,
-				};
+				let contract_address = self.contract_address;
+				let permissions = self.contract.functions()
+					.allowed_tx_types()
+					.call(sender, &|data| client.call_contract(BlockId::Hash(*parent_hash), contract_address, data))
+					.map(|p| p.low_u32())
```

Comment:
Where does this come from?

Reply:
native_contracts_generator was implicitly doing these conversion, when type was `u32`. `ethabi_derive` does not do that